### ISSUE TITLE
New API for notebook services

### ISSUE THREAD

**Issue by mohammad-sdsc** [SwissDataScienceCenter/renku-notebooks#186]
As a client of notebook services, I want to be able to have a REST API to be able to perform CRUD operations on notebook servers using `namespace`, `project`, `branch`, and `commit`.

**Comment by mohammad-sdsc** [comment]
After discussion with @rokroskar and @lorenzo-cavazzi, we agreed on the following API:
```
POST /servers - required payload: namespace, project, commit-sha, branch --> creates /servers/<server-id>
GET /servers (optional query parameters: namespace, project, commit-sha, branch)
GET /servers/<server-id>
DELETE /servers/<server-id>
```
```
GET /logs/<server-id>
```

**Comment by rokroskar** [comment]
closed by #188
